apps/explorer/src/ui/Disclosure.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Disclosure as LibDisclosure } from '@headlessui/react';
+import { ChevronDown24 } from '@mysten/icons';
+import { cva } from 'class-variance-authority';
+
+import type { ReactNode } from 'react';
+
+export type DisclosureProps = {
+    defaultOpen?: boolean;
+    title: ReactNode;
+    children: ReactNode;
+    variant: 'inline' | 'module';
+};
+
+const disclosureStyles = cva('', {
```

Comment:
I was pretty torn on what to do here -- the disclosures in our designs aren't super cohesive but I see roughly three types - Package Module  / Object Property / Inline. Considered building them all out separately, but decided against it for now. Happy to make changes here!

Reply:
will do!